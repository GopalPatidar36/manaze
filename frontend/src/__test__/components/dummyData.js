import {
  GET_ALL_TICKET,
  GET_CURRENT_USER,
  GET_TICKET,
  GET_CURRENT_USER_TICKET,
  DELETE_TICKET,
  UPDATE_USER,
  CREATE_TICKET,
  UPDATE_TICKET,
} from "../../Query/index";

export const login = {
  result: {
    data: {
      login: {
        token:
          "dummy.token.Ca1Ab6lXUn0v_nGvBnFNJDzd5SWr6hL-uEHukKW6IVeGA764MIH7pLsbrknBczubbfRedpgoOWeLp-e9jnOPFA3bjPoEYX8wTH60RUQtOiZv9ttBrxVXcpg4cRAuRCRsxyHp79riJyo8WBP_Jlwa9gnhj0vweazQeFwsq-cO1VS7EoCp_79vbr8draenM_Ly0F2ZDxxbyMguigqFYJN_lGO4V14aYhhVVZyqj2AVnp24NwdE258PH93fWscif2RE8Lxw344zoLx73S4bLo4YU9aq0Z7ZAwoATTKvC95_OIhhNFJ5QYx2HESuNYDa1Arfn4p0dqX1x9Q3ZQ1OsgMV0N7aSxRu09zwl_IL4B-3Grq2GMgFNW_icR46tSerV8wih2tZV6e18hFojtYu7I6LKWZlOYyTw2W3ZF3gZvCTjs__Gj0ZRYZJSpw1vTtxC0iR163XWDShYtWY6Rem94pSjmK_q923qe49GEMDE1F7NTC65e_kOQZwjMbu8AINIbx8uUlXei0LhdTCDXyQgUxpIjh8bTkysClVK5ZMk81w4zIGvd5h1t_4z3w2ts53iEfd_EiwJDTkxbY9GAsV5ijKJ-CQUbzPxuVuX_MTmi3kxarIdBU179EWA2cln9YMXYqWxHL2NLW0kHsoTlILzs89Mo7dvQ_Py-YyDZsHSGpNJIE",
        user: {
          uid: "236139b2-e7a5-49b8-9a56-f1b71cb12653",
          userEmail: "ramp@gmail.com",
          password: null,
          firstName: "ram",
          lastName: "Patidar",
          __typename: "RegistrationType",
        },
        __typename: "LoginTypes",
      },
    },
  },
};

export const getCurrentUSER = {
  request: { query: GET_CURRENT_USER },
  result: {
    data: {
      me: {
        __typename: "userFields",
        id: 11,
        uid: "236139b2-e7a5-49b8-9a56-f1b71cb12653",
        userEmail: "ramp@gmail.com",
        firstName: "ram",
        lastName: "Patidar",
        fullName: "ram Patidar",
        role: "member",
        createdAt: "2024-07-20T12:00:00Z",
        updatedAt: "2024-07-20T12:30:00Z",
      },
    },
  },
};

export const updateUser = {
  request: {
    query: UPDATE_USER,
    variables: { uid: "236139b2-e7a5-49b8-9a56-f1b71cb12653", firstName: "Gopal", lastName: "Patidar" },
  },
  result: {
    data: {
      updateUser: {
        __typename: "userFields",
        id: 11,
      },
    },
  },
};

export const createTicket = {
  request: {
    query: CREATE_TICKET,
    variables: {
      title: "dummpy title for test",
      description: "dummpy description for test",
      priority: "LOW",
      status: "INPROGRESS",
    },
  },
  result: { data: { createTicket: { id: 12 } } },
};

export const updateTicket = {
  request: {
    query: UPDATE_TICKET,
    variables: {
      id: 12,
      title: "update title for test",
      description: "update description for test",
      priority: "HIGH",
      status: "CLOSED",
    },
  },
  result: { data: { updateTicket: { id: 12 } } },
};

export const getUpdatedTicket = {
  request: {
    query: GET_TICKET,
    variables: { id: 12 },
  },
  result: {
    data: {
      ticketByID: {
        __typename: "Ticket",
        id: 12,
        title: "update title for test",
        description: "update description for test",
        priority: "HIGH",
        status: "CLOSED",
        createdAt: "234",
        updatedAt: "234",
        ticketUsersDetails: [
          {
            uid: "qalk-12321-w2dn",
            firstName: "gopal",
            lastName: "patidar",
            fullName: "gopal patidar",
          },
        ],
      },
    },
  },
};

export const getTicket = {
  request: {
    query: GET_TICKET,
    variables: { id: 12 },
  },
  result: {
    data: {
      ticketByID: {
        __typename: "Ticket",
        id: 12,
        title: "dummpy title for test",
        description: "dummpy description for test",
        priority: "LOW",
        status: "OPEN",
        createdAt: "234",
        updatedAt: "234",
        ticketUsersDetails: [
          {
            uid: "qalk-12321-w2dn",
            firstName: "gopal",
            lastName: "patidar",
            fullName: "gopal patidar",
          },
        ],
      },
    },
  },
};

export const deleteTicket = {
  request: {
    query: DELETE_TICKET,
    variables: { id: 12 },
  },
  result: { data: { deleteTicket: { id: 12 } } },
};

export const getAllTicket = {
  request: {
    query: GET_ALL_TICKET,
    variables: { offset: 0 },
  },
  result: {
    data: {
      ticketList: {
        __typename: "TicketList",
        count: 8,
        rows: [
          {
            __typename: "Ticket",
            id: 1,
            title: "create a ticket update  -61",
            description:
              "the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   ",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721052035000",
            createdAt: "1718715380000",
          },
          {
            __typename: "Ticket",
            id: 2,
            title:
              "ticket-702 testing is able to create a second number ticket  testing is able to create a second number ticket  testing is able to create a second number ticket testing is able to create a second number ticket  testing is able to create a second number ticket ",
            description: "testing is replace to create a second number ticket ",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721110760000",
            createdAt: "1718716225000",
          },
          {
            __typename: "Ticket",
            id: 5,
            title: "ticket-5-update",
            description: "1ticket-5-updated",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721040503000",
            createdAt: "1718716294000",
          },
          {
            __typename: "Ticket",
            id: 16,
            title: "ticket-20012",
            description: "ticket-20  update discription ",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721049603000",
            createdAt: "1718716667000",
          },
          {
            __typename: "Ticket",
            id: 17,
            title: "addTicket-17-12",
            description: "addTicket-17",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
          {
            __typename: "Ticket",
            id: 20,
            title: "title-20",
            description: "description-20",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
          {
            __typename: "Ticket",
            id: 21,
            title: "title-21",
            description: "description-21",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
          {
            __typename: "Ticket",
            id: 22,
            title: "title-22",
            description: "description-22",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
        ],
      },
    },
  },
};

export const getCurrentUSERTicket = {
  request: {
    query: GET_CURRENT_USER_TICKET,
    variables: { offset: 0 },
  },
  result: {
    data: {
      currentUserTicket: {
        __typename: "TicketList",
        count: 5,
        rows: [
          {
            __typename: "Ticket",
            id: 1,
            title: "current user a ticket update  -61",
            description:
              "the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   the update is from the ticket details page   ",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721052035000",
            createdAt: "1718715380000",
          },
          {
            __typename: "Ticket",
            id: 2,
            title:
              "ticket-702 testing is able to create a second number ticket  testing is able to create a second number ticket  testing is able to create a second number ticket testing is able to create a second number ticket  testing is able to create a second number ticket ",
            description: "testing is replace to create a second number ticket ",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721110760000",
            createdAt: "1718716225000",
          },
          {
            __typename: "Ticket",
            id: 5,
            title: "ticket-5-update",
            description: "1ticket-5-updated",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721040503000",
            createdAt: "1718716294000",
          },
          {
            __typename: "Ticket",
            id: 16,
            title: "ticket-20012",
            description: "ticket-20  update discription ",
            priority: "LOW",
            status: "OPEN",
            updatedAt: "1721049603000",
            createdAt: "1718716667000",
          },
          {
            __typename: "Ticket",
            id: 17,
            title: "addTicket-17-12",
            description: "addTicket-17",
            priority: "LOW",
            status: "INPROGRESS",
            updatedAt: "1721052088000",
            createdAt: "1718716708000",
          },
        ],
      },
    },
  },
};
