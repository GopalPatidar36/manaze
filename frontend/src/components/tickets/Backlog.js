import React, { useState } from "react";
import { getTicket } from "../../redux/slices/backlogTickets";
import DataTable from "../DataTable";
import AddTicket from "./addTicket";

const Backlog = () => {
  const headers = ["title", "description", "status", "createdAt", "updatedAt"];
  const [modal, setModal] = useState(false);

  const openModel = () => {
    setModal((item)=>!item);
  };

  return (
    <div className="backlog">
      <button className="createButton" type="submit" onClick={() => openModel()}>
        Create Issue
      </button>
      <DataTable headers={headers} api={getTicket} slice="backlog" />
      {modal && <AddTicket closeModal={openModel}/>}
    </div>
  );
};
export default Backlog;
