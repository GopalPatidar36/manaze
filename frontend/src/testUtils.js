// src/testUtils.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { Provider } from "react-redux";
import { MockedProvider } from "@apollo/client/testing";
import { BrowserRouter as Router } from "react-router-dom";
import store from "./store";

function render(ui, { preloadedState, apolloMocks, ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <MockedProvider mocks={apolloMocks} addTypename={false}>
          <Router>{children}</Router>
        </MockedProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
