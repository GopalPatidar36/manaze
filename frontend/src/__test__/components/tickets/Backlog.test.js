import { screen, waitFor, act } from "@testing-library/react";
import Backlog from "../../../components/tickets/Backlog";
import { render } from "../../../testUtils";
import { getAllTicket } from "../dummyData";

const mocks = [getAllTicket];

describe("Testing with Backlog data", () => {
  let components;
  beforeEach(async () => {
    await act(async () => {
      components = render(<Backlog />, { apolloMocks: mocks });
    });
  });
  test("Check backlog text", async () => {
    await waitFor(() => {
      expect(screen.getByText(/Backlog Issue/)).toBeInTheDocument();
    });
  });

  test("Check Create text", async () => {
    await waitFor(() => {
      expect(screen.getByText(/Create Issue/)).toBeInTheDocument();
    });
  });
  test("Check ticket discription text", async () => {
    await waitFor(() => {
      const elements = screen.getAllByText(/addTicket-17/);
      expect(elements[0]).toBeInTheDocument();
    });
  });
  it("should match dom snapshot:-Backlog", async () => {
    await waitFor(() => {
      expect(components.asFragment()).toMatchSnapshot();
    });
  });
});
