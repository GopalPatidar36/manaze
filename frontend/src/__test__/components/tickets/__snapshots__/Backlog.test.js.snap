// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with Backlog data should match dom snapshot:-Backlog 1`] = `
<DocumentFragment>
  <div
    class="backlog"
  >
    <div
      class="backlog2"
    >
      <h3
        class=""
      >
        Backlog Issue
      </h3>
      <button
        type="submit"
      >
        Create Issue
      </button>
    </div>
    <div
      class="DataTable"
    >
      <div
        style="display: flex; flex-wrap: wrap; margin-top: 5px;"
      >
        <input
          class="searchInput"
          placeholder="Search Ticket"
          value=""
        />
        <div
          style="flex-direction: column; display: flex; margin-left: 10px;"
        >
          <label
            class="label"
          >
            Priority
          </label>
          <select
            class="selectOption"
            id="country"
          >
            <option
              value=""
            >
              None
            </option>
            <option
              value="LOW"
            >
              Low
            </option>
            <option
              value="MEDIUM"
            >
              Medium
            </option>
            <option
              value="HIGH"
            >
              High
            </option>
          </select>
        </div>
        <div
          style="flex-direction: column; display: flex; margin-left: 10px;"
        >
          <label
            class="label"
          >
            Status
          </label>
          <select
            class="selectOption"
            id="country"
          >
            <option
              value=""
            >
              None
            </option>
            <option
              value="OPEN"
            >
              Open
            </option>
            <option
              value="INPROGRESS"
            >
              In Progress
            </option>
            <option
              value="CLOSED"
            >
              Close
            </option>
          </select>
        </div>
      </div>
      <div
        style="overflow-x: auto;"
      >
        <table>
          <tbody>
            <tr>
              <th
                class="tableHeader"
                id="title"
              >
                <span
                  style="cursor: pointer;"
                >
                  Title
                </span>
              </th>
              <th
                class="tableHeader"
                id="description"
              >
                <span
                  style="cursor: pointer;"
                >
                  Description
                </span>
              </th>
              <th
                class="tableHeader"
                id="priority"
              >
                <span
                  style="cursor: pointer;"
                >
                  Priority
                </span>
              </th>
              <th
                class="tableHeader"
                id="status"
              >
                <span
                  style="cursor: pointer;"
                >
                  Status
                </span>
              </th>
              <th
                class="tableHeader"
                id="createdAt"
              >
                <span
                  style="cursor: pointer;"
                >
                  CreatedAt
                </span>
              </th>
              <th
                class="tableHeader"
                id="updatedAt"
              >
                <span
                  style="cursor: pointer;"
                >
                  UpdatedAt
                </span>
              </th>
              <th
                id="actionHeader"
              />
            </tr>
            <tr
              class="tableDataRow"
            >
              <td
                id="title"
              >
                create a ticket update  -61
              </td>
              <td
                id="description"
              >
                the update is from the ticket details page   the u...
              </td>
              <td
                id="priority"
              >
                LOW
              </td>
              <td
                id="status"
              >
                OPEN
              </td>
              <td
                id="createdAt"
              >
                18/6/2024
              </td>
              <td
                id="updatedAt"
              >
                15/7/2024
              </td>
              <td
                id="action"
                style="width: 1px; z-index: 5;"
              >
                ⋮
                <div
                  class="actionInDatatable"
                >
                  <button
                    class="actionButton"
                  >
                    Edit
                  </button>
                  <button
                    class="actionButton"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr
              class="tableDataRow"
            >
              <td
                id="title"
              >
                ticket-702 testing is able to create a second numb...
              </td>
              <td
                id="description"
              >
                testing is replace to create a second number ticke...
              </td>
              <td
                id="priority"
              >
                LOW
              </td>
              <td
                id="status"
              >
                INPROGRESS
              </td>
              <td
                id="createdAt"
              >
                18/6/2024
              </td>
              <td
                id="updatedAt"
              >
                16/7/2024
              </td>
              <td
                id="action"
                style="width: 1px; z-index: 5;"
              >
                ⋮
                <div
                  class="actionInDatatable"
                >
                  <button
                    class="actionButton"
                  >
                    Edit
                  </button>
                  <button
                    class="actionButton"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr
              class="tableDataRow"
            >
              <td
                id="title"
              >
                ticket-5-update
              </td>
              <td
                id="description"
              >
                1ticket-5-updated
              </td>
              <td
                id="priority"
              >
                LOW
              </td>
              <td
                id="status"
              >
                OPEN
              </td>
              <td
                id="createdAt"
              >
                18/6/2024
              </td>
              <td
                id="updatedAt"
              >
                15/7/2024
              </td>
              <td
                id="action"
                style="width: 1px; z-index: 5;"
              >
                ⋮
                <div
                  class="actionInDatatable"
                >
                  <button
                    class="actionButton"
                  >
                    Edit
                  </button>
                  <button
                    class="actionButton"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr
              class="tableDataRow"
            >
              <td
                id="title"
              >
                ticket-20012
              </td>
              <td
                id="description"
              >
                ticket-20  update discription 
              </td>
              <td
                id="priority"
              >
                LOW
              </td>
              <td
                id="status"
              >
                OPEN
              </td>
              <td
                id="createdAt"
              >
                18/6/2024
              </td>
              <td
                id="updatedAt"
              >
                15/7/2024
              </td>
              <td
                id="action"
                style="width: 1px; z-index: 5;"
              >
                ⋮
                <div
                  class="actionInDatatable"
                >
                  <button
                    class="actionButton"
                  >
                    Edit
                  </button>
                  <button
                    class="actionButton"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
            <tr
              class="tableDataRow"
            >
              <td
                id="title"
              >
                addTicket-17-12
              </td>
              <td
                id="description"
              >
                addTicket-17
              </td>
              <td
                id="priority"
              >
                LOW
              </td>
              <td
                id="status"
              >
                INPROGRESS
              </td>
              <td
                id="createdAt"
              >
                18/6/2024
              </td>
              <td
                id="updatedAt"
              >
                15/7/2024
              </td>
              <td
                id="action"
                style="width: 1px; z-index: 5;"
              >
                ⋮
                <div
                  class="actionInDatatable"
                >
                  <button
                    class="actionButton"
                  >
                    Edit
                  </button>
                  <button
                    class="actionButton"
                  >
                    Delete
                  </button>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="pagginationContainer"
      >
        <span>
          <svg
            class="forwadAndBackWordIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.5 440.6c9.5 7.9 22.8 9.7 34.1 4.4s18.4-16.6 18.4-29V96c0-12.4-7.2-23.7-18.4-29s-24.5-3.6-34.1 4.4L288 214.3V256v41.7L459.5 440.6zM256 352V256 128 96c0-12.4-7.2-23.7-18.4-29s-24.5-3.6-34.1 4.4l-192 160C4.2 237.5 0 246.5 0 256s4.2 18.5 11.5 24.6l192 160c9.5 7.9 22.8 9.7 34.1 4.4s18.4-16.6 18.4-29V352z"
            />
          </svg>
        </span>
        <button
          class="paginationPoint"
          style="background-color: blue; color: white;"
        >
          1
        </button>
        <span>
          <svg
            class="forwadAndBackWordIcon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M52.5 440.6c-9.5 7.9-22.8 9.7-34.1 4.4S0 428.4 0 416V96C0 83.6 7.2 72.3 18.4 67s24.5-3.6 34.1 4.4L224 214.3V256v41.7L52.5 440.6zM256 352V256 128 96c0-12.4 7.2-23.7 18.4-29s24.5-3.6 34.1 4.4l192 160c7.3 6.1 11.5 15.1 11.5 24.6s-4.2 18.5-11.5 24.6l-192 160c-9.5 7.9-22.8 9.7-34.1 4.4s-18.4-16.6-18.4-29V352z"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
